dattri/func/ihvp.py

```diff
@@ -877,29 +893,38 @@ def ihvp_at_x_datainf(
         A function that takes a tuple `v` and returns the tuple of IHVPs of the Hessian
         of `func` and `v`.
     """
+
     def _per_sample_grad(*args) -> torch.Tensor:
         return grad(func, argnums=argnums)(*args)
 
-    def _single_datainf_ihvp(v: torch.Tensor,
-                                     grad: torch.Tensor,
-                                     regularization: float) -> torch.Tensor:
-        coef = (v.T @ grad) / (regularization + torch.sum(grad ** 2))
+    def _single_datainf_ihvp(
```

Comment:
It seems like there is a redundancy for this function and the above `_ihvp_datainf_func`?
These codes look really similar.

Reply:
will add a todo for @charles-pyj